src/index.ts

```diff
@@ -278,24 +281,24 @@ function setByPath<
     throw new Error('Path is empty')
   }
 
-  // eslint-disable-next-line @typescript-eslint/no-unsafe-assignment
-  const parentObject = pathArray.reduce((current: any, pathPart) => {
-    if (typeof current !== 'object' || !hasOwnProperty.call(current, pathPart)) {
+  const parentObject = pathArray.reduce((current: unknown, pathPart) => {
+    if (
+      typeof current !== 'object' ||
+      current === null ||
+      !hasOwnProperty.call(current, pathPart)
+    ) {
       throw new Error(`Property ${pathPart} is undefined`)
     }
 
-    // eslint-disable-next-line @typescript-eslint/no-unsafe-assignment, @typescript-eslint/no-unsafe-member-access
-    const next = current?.[pathPart]
+    const next: unknown = (current as SafeObject)[pathPart]
 
     if (next === undefined || next === null) {
       throw new Error(`Property ${pathPart} is undefined`)
     }
 
-    // eslint-disable-next-line @typescript-eslint/no-unsafe-return
     return next
-  }, object)
+  }, object) as SafeObject
 
-  // eslint-disable-next-line @typescript-eslint/no-unsafe-member-access
   parentObject[lastKey] = value
```

Comment:
Btw I noticed that here we don't necessarly have an object either

We can have anything except undefined or null, does it make sense to throw an exception if we try to assign to a non object like a string or a number (javascript will accept it because you can pretty much assign to anything)

Reply:
I think we should because then it can give really unexpected results

![image](https://github.com/clickbar/dot-diver/assets/6115458/8cdb12e3-10b1-4c84-94c6-6dc8d155df31)

Reply:
Oh that seems cursed @djfhe what do you think?

Reply:
It seems to be handled by the HasOwnProperty check, I'll send a PR that adds the tests